### PR TITLE
Fix some golint failures of pkg/kubectl/cmd/wait

### PR DESCRIPTION
**Pull request by khrd**
**What type of PR is this?**

 /kind cleanup

**What this PR does / why we need it**:
Fix some golint failures of pkg/kubectl/cmd/wait golint

**Which issue(s) this PR fixes**:

Ref #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Review by mattjmcnaughton — CHANGES_REQUESTED** [review]
thanks for your pr :)

At a high level, do you know if there is an overall plan to address `stuttering` lint failures in `pkg/kubectl/cmd`? The majority of files I looked at follow the pattern `WaitOptions` as opposed to just `Options`. I'm comfortable using the linter recommended `Options`, but just want to make sure there's a plan to apply this to all of `pkg/kubectl/cmd` and that the use of `WaitOptions` wasn't intentional despite the lint failure.

/ok-to-test

lmk if you need any help signing the CLA :)

File: pkg/kubectl/cmd/wait/wait.go [hunk 1]
```diff
@@ -206,9 +206,9 @@ type ResourceLocation struct {
 // UIDMap maps ResourceLocation with UID
 type UIDMap map[ResourceLocation]types.UID
 
-// WaitOptions is a set of options that allows you to wait.  This is the object reflects the runtime needs of a wait
+// Options is a set of options that allows you to wait.  This is the object reflects the runtime needs of a wait
 // command, making the logic itself easy to unit test with our existing mocks.
-type WaitOptions struct {
+type Options struct {
```
Comment (mattjmcnaughton) [inline pkg/kubectl/cmd/wait/wait.go]: Whoops, didn't save this comment before submitting my review :)

`WaitOptions` is used in files outside of this one... please make sure you update all references.

Comment (palnabarun) [reply pkg/kubectl/cmd/wait/wait.go]: I think it is not recommended to fix this golint error. There have been pending discussions going on for this change at various places - https://github.com/kubernetes/kubernetes/pull/70682#discussion_r233642556 , https://github.com/kubernetes/kubernetes/pull/78634 .

This needs to be sorted out and written in the contribution guidelines.

Comment (khrd) [reply pkg/kubectl/cmd/wait/wait.go]: @mattjmcnaughton @palnabarun  thanks for your reviews and comments :)

> We don't need to blindly follow all golint suggestions, I consider this just that. That's why I think we should leave the current naming scheme as is. -  #70682 (comment)

hmm.. I think so.

**Pull request closed without merging**